lib/3.6/bundles.cf

```diff
@@ -416,3 +416,54 @@ bundle agent run_ifdefined(namespace, mybundle)
     verbose_mode::
       "$(this.bundle): found matching bundles $(bundlesfound) for namespace '$(namespace)' and bundle '$(mybundle)'";
 }
+bundle agent sysctl(settings_array)
+# @brief bundle to set and load sysctl settings
+# @param name of array containing sysctl settings
+#
+# This bundle is a wrapper around set_variable_values 
+# targetting /etc/sysctl.conf. It runs sysctl -p after
+# editing sysctl.conf, to load the new values.
+# Contributed by Aleksey Tsalolikhin, 2 July 2015
+#
+# **Example:**
+#
+# ```cf3
+# bundle agent example
+# {
+#   vars:
+#       "sysctl_settings"
+#         data => parsejson('{
+#                              "net.core.rmem_max"     :  "4194304",
+#                              "net.core.rmem_default" : "4194304",
+#                              "vm.swappiness"         : "5",
+#                            }');
+#
+#   methods:
+#       "Edit and reload /etc/sysctl.conf" usebundle => sysctl("$(this.bundle).sysctl_settings");
+# }
+#
+# ```
+{
+  vars:
+      "sysctl_conf" string => "/etc/sysctl.conf";
+
+    DEBUG|DEBUG_sysctl|verbose_mode::
+      "keys" slist => getindices("$(settings_array)");
+
+  files:
+      "$(sysctl_conf)"
+        edit_line     => set_variable_values("$(settings_array)"),
+        edit_defaults => backup_timestamp,
+        classes       => if_repaired("reinit_sysctl");
```

Comment:
if_repaired produces a globaly scoped class. Could you please use a classes body that defines a bundle scoped class? I don't see any reason to signal outside this bundle.

Perphas: 

classes => scoped_classes_generic("bundle", "sysctl_conf")

And then to guard the reload command:

sysctl_conf_repaired::